src/input.rs

```diff
@@ -0,0 +1,49 @@
+use std::fmt;
+use crate::event::filter::{EventFilter};
+use crate::{csi, Command, Result};
+use crate::event::{Event, InternalEvent, read_internal};
+
+pub fn read() -> Result<Input> {
+    match read_internal(&EventFilter)? {
+        InternalEvent::Event(event) => Ok(Input::Event(event)),
+        InternalEvent::Input(input) => Ok(input),
+        #[cfg(unix)]
+        _ => unreachable!()
+    }
+}
+
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Debug, PartialOrd, PartialEq, Eq, Clone, Hash)]
+pub enum Input {
+    Event(Event),
+    Paste(String),
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct EnableBracketedPaste;
```

Comment:
I put these commands in here as they're only applicable if you're using the read function in this module. It's probably worth panicing or something like that if Paste event is emitted when you've called the `event::read` function as that means you're mixing the systems. Not the greatest API, but I don't see a way to do better without breaking compatibility.

Reply:
Can you explain what the problem is? Like why would you need panicking or change event when you've called read?

Reply:
If you've used EnableBracketedPaste and then you call `event::read` instead of `input::read`, you'll never get the paste input you requested. It's programmer error to mix the two. I want some way to let them know there's a mismatch instead of having it silently fail. Maybe there's a better way?